feat: custom protocol name (#962)
libp2p/js-libp2p: doc/CONFIGURATION.md

[NAT-PMP](http://miniupnp.free.fr/nat-pmp.html) is a feature of some modern routers which performs a similar job to UPnP. NAT-PMP is disabled by default, if enabled libp2p will try to use NAT-PMP and will fall back to UPnP if it fails.

#### Configuring protocol name

> Changing protocol name is used to isolate default public network (IPFS) for custom purposes. 

Reviewer: ```suggestion
Changing the protocol name prefix can used to isolate default public network (IPFS) for custom purposes. 
```

```js
const node = await Libp2p.create({
  config: {

>     protocol: 'ipfs' // default

Reviewer 2: Would suggest calling this protocolPrefix.

  }
})
/*
protocols: [
  "/ipfs/id/1.0.0", // identify service protocol (if we have multiplexers)

>   "/ipfs/id/push/1.0.0", // identify service protocol (if we have multiplexers)

Reviewer: ```suggestion
  "/ipfs/id/push/1.0.0", // identify service push protocol (if we have multiplexers)
```

>   "/ipfs/ping/1.0.0", // built-in protocol

Reviewer: ```suggestion
  "/ipfs/ping/1.0.0", // built-in ping protocol
```
